CMakeLists.txt

```diff
@@ -72,6 +72,13 @@ if(USE_R35)
 endif(USE_R35)
 
 if(BUILD_FOR_R)
+    list(APPEND CMAKE_MODULE_PATH "${lightgbm_SOURCE_DIR}/cmake/modules")
```

Comment:
I saw this [all over xgboost's cmake scripts](https://github.com/dmlc/xgboost/search?q=xgboost_source_dir&unscoped_q=xgboost_source_dir). I thought it might be that this is defined for `<project>_SOURCE_DIR` where `<project>` is a name you gave to `PROJECT()`, like we do at https://github.com/microsoft/LightGBM/blob/master/CMakeLists.txt#L7

The documentation on `SOURCE_DIR` is pretty minimal https://cmake.org/cmake/help/v3.16/prop_dir/SOURCE_DIR.html 😅

Reply:
found the docs! https://cmake.org/cmake/help/v3.0/variable/PROJECT-NAME_SOURCE_DIR.html

Reply:
there are [a bunch of these](https://cmake.org/cmake/help/v3.0/manual/cmake-variables.7.html)

![image](https://user-images.githubusercontent.com/7608904/76875827-1606e380-683f-11ea-89f1-f65106ec32ae.png)

Reply:
I think we should stick to the currently using [`PROJECT_SOURCE_DIR`](https://cmake.org/cmake/help/v3.0/variable/PROJECT_SOURCE_DIR.html), as we have only one project.

Reply:
that is very reasonable! I'll change it